### ISSUE TITLE
useSlider reads one value behind input value

### ISSUE THREAD

**Issue by courtneylinder** [adobe/react-spectrum#5583]
### Provide a general summary of the issue here

When testing useSlider in VoiceOver for both Chrome and Safari, it seems to be reading one value behind the actual input value of the slider. Instead of reading the updated value it reads the previous one.

<img width="644" alt="Screenshot 2023-12-15 at 9 22 20 AM" src="https://github.com/adobe/react-spectrum/assets/14795226/10e0eec2-33ce-4093-b253-e4e3a1e759bb">
<img width="679" alt="Screenshot 2023-12-15 at 9 22 48 AM" src="https://github.com/adobe/react-spectrum/assets/14795226/fb9e5fcc-3f91-4dfe-a4c1-2d0715252c4a">

### 🤔 Expected Behavior?

That the Slider reads the same value as what would be displayed to sighted users.

### 😯 Current Behavior

The VoiceOver reads previous value for input range instead of updated value. i.e. Instead of saying 21, it reads 20. 

### 🖥️ Steps to Reproduce

Can test by going to any single sliders on the docs, turning on VoiceOver for Chrome or Safari, and moving the slider. Gets really funky when you start to go forward and backwards too.
https://react-spectrum.adobe.com/react-aria/useSlider.html#single-thumb

### Version

react-aria 3.29.1, react-stately 3.27.1

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari

### What operating system are you using?

macOS Sonoma 14.1.1

**Comment by snowystinger** [comment]
Thanks for reporting. Chiming in from my phone, I can't reproduce on iOS 17.1.2. I or one of my teammates will try desktop soon. Just another data point with iOS.

Does it work any better if you use the VO key? (Ctrl+Opt+arrow) or is that what you were using already?

**Comment by ryo-manba** [comment]
I've tested the useSlider issue with VoiceOver on my setup and couldn't replicate the problem. 
Here's what I used:
- MacOS Sonoma 14.1.1
- Chrome Version 120.0.6099.71
- Safari Version 17.1 (19616.2.9.11.7)

In these environments, VoiceOver read the slider values correctly. Not sure if it's an environment-specific issue or something else. Could you provide more information or specific steps on how you encountered this issue? It might help in narrowing down the cause.


### Safari

https://github.com/adobe/react-spectrum/assets/76232929/633ccfb7-12f3-4ab7-a8ef-d70d1a3a6e56

### Chrome
https://github.com/adobe/react-spectrum/assets/76232929/a6a4242d-4b53-448d-ae56-c202846b3653

**Comment by courtneylinder** [comment]
https://github.com/adobe/react-spectrum/assets/14795226/50defe41-fdef-43a9-bf72-cc069cda952b

I don't do anything special I don't think. I turn on voice over, navigate to the slider, VO down arrow into the slider and start moving it.

**Comment by reidbarber** [comment]
Just tried on Chrome 120 MacOS 14.1.2 Intel, and Safari 17.1 MacOS 14.1 M2, and couldn't reproduce the issue.

Using Option+Left/Right to navigate.


https://github.com/adobe/react-spectrum/assets/8961049/0e57b820-67d7-4922-8ec4-535c17011f73


https://github.com/adobe/react-spectrum/assets/8961049/594e24ad-c3dc-4672-bc26-ae2eb4af4d77

**Comment by courtneylinder** [comment]
Yeah if you hold option it works but what about quick nav arrow keys? As that is also a viable way to navigate

**Comment by reidbarber** [comment]
I believe with Quick Nav turned on, you can simultaneously press the left and right arrow keys to enter the slider, then use left and right arrow keys to adjust the slider value, then to exit the slider you can simultaneously press the left and right arrow keys again.

**Comment by courtneylinder** [comment]
Yes exactly, that's how I use Quick Nav. When having Quick Nav on is when I come across these issues. Did you try it with quick nav on?

**Comment by yihuiliao** [comment]
I'm able to reproduce on MacOS 13.5.2, Chrome 119 when I shift + control + option + down when I'm on the slider, then use the left and right arrow key to adjust the values.

**Comment by ryo-manba** [comment]
I have observed the same behavior on the [Color Viewer Slider Example | APG | WAI | W3C](https://www.w3.org/WAI/ARIA/apg/patterns/slider/examples/slider-color-viewer/).
This suggests that the issue may not lie with the useSlider itself, but rather with some aspect of macOS's reading functionality.

**Comment by snowystinger** [comment]
Unable to reproduce in 148.0.7778.216
I'm going to close as stale/VO bug that isn't our issue.

If it's still happening to people, we'll probably need to ask that you check against the APG / native controls for it. If it's not an issue there but is for us, then we can dive in and compare again now that some time has passed.
